pkg/directorysync/directorysync_test.go

```diff
@@ -296,3 +296,128 @@ func getDirectoryTestHandler(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	w.Write(body)
 }
+
+func TestPrimaryEmail(t *testing.T) {
+	tests := []struct {
+		scenario string
+		user     User
+		expected string
+		err      bool
+	}{
+		{
+			scenario: "One primary email returns primary email",
+			user: User{
+				ID:        "directory_user_id",
+				FirstName: "WorkOS",
+				LastName:  "Testz",
+				Emails: []UserEmail{
+					UserEmail{
+						Primary: true,
+						Type:    "work",
+						Value:   "primaryemail@foo-corp.com",
+					},
+				},
+				Groups: []UserGroup{
+					UserGroup{
+						Object: "user_group_object",
+						ID:     "directory_group_123",
+						Name:   "Group Name",
+					},
+				},
+				State:            Active,
+				RawAttributes:    json.RawMessage(`{"foo":"bar"}`),
+				CustomAttributes: json.RawMessage(`{"foo":"bar"}`),
+			},
+			expected: "primaryemail@foo-corp.com",
+		},
+		{
+			scenario: "Multiple primary emails returns the first primary email",
+			user: User{
+				ID:        "directory_user_id",
+				FirstName: "WorkOS",
+				LastName:  "Testz",
+				Emails: []UserEmail{
+					UserEmail{
+						Primary: true,
+						Type:    "work",
+						Value:   "firstprimaryemail@foo-corp.com",
+					},
+					UserEmail{
+						Primary: true,
+						Type:    "work",
+						Value:   "primaryemail@foo-corp.com",
+					},
+				},
+				Groups: []UserGroup{
+					UserGroup{
+						Object: "user_group_object",
+						ID:     "directory_group_123",
+						Name:   "Group Name",
+					},
+				},
+				State:            Active,
+				RawAttributes:    json.RawMessage(`{"foo":"bar"}`),
+				CustomAttributes: json.RawMessage(`{"foo":"bar"}`),
+			},
+			expected: "firstprimaryemail@foo-corp.com",
+		},
+		{
+			scenario: "No primary emails returns null",
```

Comment:
It looks like we're returning an error rather than `null` here. Can we rename these tests to indicate an error is returned as well as an empty string?